src/adapters/store/postgres.ts

```diff
@@ -109,50 +103,54 @@ export function createPgStoreAdapter(
   }
 
   async function lastEventOf<E extends Event<any, any>>(
-    eventType: EventNamespaceAndType,
+    ns: EventNamespace,
+    ty: EventType
   ): Promise<Option<E>> {
-    logger.trace({ eventType }, 'eventStoreLastEventOfResult');
+    logger.trace({ ns, ty }, "eventStoreLastEventOf");
 
     const start = Date.now();
 
     return pool
       .query(
-        `select * from events where data->>'type' = $1 order by context->>'time' desc limit 1`,
-        [eventType],
+        `select * from events
+        where data->>'event_namespace' = $1
+        and data->>'event_type' = $2
+        order by context->>'time' desc limit 1`,
+        [ns, ty]
       )
       .then((results) => {
-        logger.trace({ eventType, time: Date.now() - start }, 'eventStoreLastEventOfResult');
+        logger.trace({ ns, ty, time: Date.now() - start }, "eventStoreLastEventOfResult");
 
         return Option.of(results.rows[0]);
       });
   }
 
   async function exists(id: Uuid): Promise<boolean> {
```

Comment:
Do we even need this function anymore?

Reply:
It's not used in the library itself so I'll remove it. I thought I might leave it in so client code can use it but that doesn't make much sense.